File: 386-lexicographical-numbers/lexicographical-numbers.py
class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        res = []
        def dfs(a : int):
            if a > n:
                return
            res.append(a)
            dfs(a*10)
            if ((a+1)%10 != 0):
                dfs(a+1)
        dfs(1)
        return res

File: 386-lexicographical-numbers/lexicographical-numbers.cpp
class Solution {
public:
    
    vector<int> lexicalOrder(int n) {
        vector<int> ans;
        function<void(int)> dfs = [&](int a) -> void {
            if (a > n) return;
            ans.emplace_back(a);
            dfs(a*10);
            if ((a+1)%10 != 0) dfs(a+1);
        };
        dfs(1);
        return ans;
    }
};